src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -0,0 +1,236 @@
+
+                               chip::app::CommandHandler * handler = handle.Get();
+                               if (handler) {
+                                   handler->AddResponse(cachedCommandPath, response);
+                                   handle.Release();
+                               }
+                           }];
+            });
+        }
+    }
+}
+
+void MTROTAProviderDelegateBridge::HandleNotifyUpdateApplied(chip::app::CommandHandler * commandObj,
+    const chip::app::ConcreteCommandPath & commandPath,
+    const chip::app::Clusters::OtaSoftwareUpdateProvider::Commands::NotifyUpdateApplied::DecodableType & commandData)
+{
+    // Make sure to hold on to the command handler and command path to be used in the completion block
+    __block chip::app::CommandHandler::Handle handle(commandObj);
+    __block chip::app::ConcreteCommandPath cachedCommandPath(
+        commandPath.mEndpointId, commandPath.mClusterId, commandPath.mCommandId);
+
+    id<MTROTAProviderDelegate> strongDelegate = mDelegate;
+    if ([strongDelegate respondsToSelector:@selector(handleNotifyUpdateApplied:completionHandler:)]) {
+        if (strongDelegate && mQueue) {
+            auto * commandParams = [[MTROtaSoftwareUpdateProviderClusterNotifyUpdateAppliedParams alloc] init];
+            ConvertToNotifyUpdateAppliedParams(commandData, commandParams);
+
+            dispatch_async(mQueue, ^{
+                [strongDelegate
+                    handleNotifyUpdateApplied:commandParams
+                            completionHandler:^(NSError * _Nullable error) {
+                                chip::app::CommandHandler * handler = handle.Get();
+                                if (handler) {
+                                    handler->AddStatus(cachedCommandPath, chip::Protocols::InteractionModel::Status::Success);
+                                    handle.Release();
+                                }
+                            }];
+            });
+        }
+    }
+}
+
+void MTROTAProviderDelegateBridge::ConvertToQueryImageParams(
+    const chip::app::Clusters::OtaSoftwareUpdateProvider::Commands::QueryImage::DecodableType & commandData,
+    MTROtaSoftwareUpdateProviderClusterQueryImageParams * commandParams)
+{
+    commandParams.vendorId = [NSNumber numberWithInt:commandData.vendorId];
+    commandParams.productId = [NSNumber numberWithInt:commandData.productId];
+    commandParams.softwareVersion = [NSNumber numberWithInt:commandData.softwareVersion];
+    auto iterator = commandData.protocolsSupported.begin();
+    NSMutableArray * protocolsSupported = [[NSMutableArray alloc] init];
+    while (iterator.Next()) {
+        chip::app::Clusters::OtaSoftwareUpdateProvider::OTADownloadProtocol protocol = iterator.GetValue();
+        [protocolsSupported addObject:[NSNumber numberWithInt:static_cast<int>(protocol)]];
+    }
+    commandParams.protocolsSupported = [protocolsSupported copy];
```

Comment:
This needs to check the iterator status, no?  Otherwise we're silently ignoring decode failures.